### PR TITLE
Components Panel

### PR DESCRIPTION
**Pull request by kof**
## Description

Implements all design changes and accessibility for components add panel

https://user-images.githubusercontent.com/52824/227720011-344b31f8-ce17-4d13-ad4d-2d506ca3bcce.mp4

## Code Review

- [ ] hi @rpominov , I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/builder/features/sidebar-left/panels/components/components.tsx [hunk 1]
```diff
@@ -1,160 +1,55 @@
...
-        icon={<meta.Icon />}
-        style={{
-          transform: `translate3d(${point.x}px, ${point.y}px, 0)`,
-        }}
-      />
-    </Flex>,
-    document.body
-  );
-};
+for (const name of getComponentNames()) {
```
Comment (TrySound) [inline apps/builder/app/builder/features/sidebar-left/panels/components/components.tsx:29]: This top level stuff may not play well with injecting components. Might be better to do inside component

File: apps/builder/app/builder/features/sidebar-left/panels/components/components.tsx [hunk 2]
```diff
@@ -163,43 +58,52 @@ export const TabContent = ({ publish, onSetActiveTab }: TabContentProps) => {
...
+                        return null;
+                      }
+                      return (
+                        <Tooltip content={meta.label} key={component}>
+                          <ComponentCard
+                            onClick={() => {
+                              onSetActiveTab("none");
+                              handleInsert(component);
+                            }}
+                            onKeyDown={(event) => {
```
Comment (TrySound) [inline apps/builder/app/builder/features/sidebar-left/panels/components/components.tsx:85]: Doesn't enter trigger click? Kinda onPress stuff. Maybe try `@react-aria/press`?

File: packages/react-sdk/src/components/component-meta.ts [hunk 3]
```diff
@@ -13,29 +13,21 @@ const WsComponentPropsMeta = z.object({
...
+export const componentCategories = [
+  "general",
+  "typography",
+  "media",
+  "forms",
+] as const;
 
 const WsComponentMeta = z.object({
+  category: z.enum(componentCategories).optional(),
+  // body - can accept other components with dnd but not listed
```
Comment (TrySound) [inline packages/react-sdk/src/components/component-meta.ts:25]: category removes the need for body to have separate type, can be removed in separate PR. Especially since we gonna have other not listed containers (fragment).

**Review by TrySound — APPROVED** [review]

File: apps/builder/app/builder/features/sidebar-left/panels/components/components.tsx (hunk 2 above)
Comment (kof) [reply apps/builder/app/builder/features/sidebar-left/panels/components/components.tsx:93]: it doesn't trigger click, probably that behavior exists only for buttons? haven't really looked into that package `@react-aria/press` maybe we should start using it all the time?

Comment (kof) [reply apps/builder/app/builder/features/sidebar-left/panels/components/components.tsx]: done

File: packages/react-sdk/src/components/component-meta.ts (hunk 3 above)
Comment (kof) [reply packages/react-sdk/src/components/component-meta.ts]: good catch, done

**Review by TrySound — APPROVED** [review]

**Comment by kof** [comment]
I will merge it now since it became quite big, I will handle all additional comments in a separate PR @rpominov @taylornowotny

**Comment by taylornowotny** [comment]
Looks awesome! 🔥🔥🔥

I asked Mark to make a separate PR for aligning the icons with the design.

**Review by rpominov — COMMENTED** [review]
Looks good!
